apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx

```diff
@@ -0,0 +1,172 @@
+import { useEffect, useState } from 'react';
+import { Form, Container, Row, Col } from 'react-bootstrap';
+import { Neighborhood, NeighborhoodsPerPage } from '@neighborhood/backend/src/types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { AxiosError } from 'axios';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+import InfiniteScroll from 'react-infinite-scroll-component';
+import styles from './NeighborhoodSearch.module.css';
+import NeighborhoodCard from '../NeighborhoodCard/NeighborhoodCard';
+import CustomBtn from '../CustomBtn/CustomBtn';
+import CreateNeighborhoodModal from '../CreateNeighborhoodModal/CreateNeighborhoodModal';
+import neighborhoodsService from '../../services/neighborhoods';
+import SpinWheel from '../SpinWheel/SpinWheel';
+import AlertBox from '../AlertBox/AlertBox';
+import { ErrorObj } from '../../types';
+
+export default function NeighborhoodSearch({
+  neighborhoods,
+  cursor,
+  isNextPage,
+}: {
+  neighborhoods: Neighborhood[];
+  cursor?: number;
+  isNextPage: boolean;
+}) {
+  const [neighborhoodList, setNeighborhoodList] = useState(neighborhoods);
+  const [hasNextPage, setHasNextPage] = useState(isNextPage);
```

Comment:
Does changing this need to trigger a re-render?

Reply:
Its value is passed to `InfiniteScroll` so I'd think yes.